sympy/physics/vector/frame.py

```diff
@@ -762,6 +762,44 @@ def orient_explicit(self, parent, dcm):
         parent._ang_vel_dict.update({self: -wvec})
         self._var_dict = {}
 
+
+    def orient_dcm(self, parent, dcm):
+        """Sets the orientation of this reference frame relative to a parent
+        reference frame using a direction cosine matrix that describes the rotation
+        from the parent to the child.
+
+        Parameters
+        ==========
+
+        parent : ReferenceFrame
+            Reference frame that this reference frame will be rotated relative
+            to.
+        dcm : Matrix, shape(3, 3)
+            Direction cosine matrix that specifies the relative rotation between
+            the two reference frames. The DCM should describe the rotation from the
+            parent to the child reference frame.
+
+        """
+
+        _check_frame(parent)
+        if not isinstance(dcm, MatrixBase):
+            raise TypeError("dcm must be a SymPy Matrix type object.")
+
+        parent_to_child_dcm = dcm
+
+        # Convert the parent-to-child DCM to child-to-parent DCM.
+        child_to_parent_dcm = parent_to_child_dcm.T
+
+        # Set the orientation of this reference frame relative to the parent
+        # reference frame using the child-to-parent DCM.
+        self._dcm(parent, child_to_parent_dcm)
```

Comment:
I would propose removing the above two statements: `parent_to_child_dcm = dcm` and `child_to_parent_dcm = ...`. As this comment is already clear enough so you can just use `self._dcm(parent, dcm.T)`. In my opinion you may also rename the function argument to `parent_to_child_dcm`, so `def orient_dcm(self, parent, parent_to_child_dcm)`.